src/environmentd/src/http/webhook.rs

```diff
@@ -50,33 +51,103 @@ pub async fn handle_webhook(
     }
     let headers = Arc::new(headers_s);
 
+    // Append to the webhook source, retrying if we race with a concurrent `ALTER SOURCE` op.
+    Retry::default()
```

Comment:
I don't think we need to limit the number of retries? Is there a problem if this is just `loop`? That way ChannelClosed will never be returned to a user, which is a pretty inscrutable error.

Reply:
Personally I'm not the biggest fan of `loop`, for example while it's unlikely, theoretically someone could start returning `ChannelClosed` from a new place in code that isn't retry-able, and we'd end up hanging the request until the external client times out.

I have an idea of how to clean up the errors a bit, I'll work on it in a follow up PR